src/legacy/core_plugins/metrics/server/lib/vis_data/response_processors/series/time_shift.js

```diff
@@ -24,15 +24,21 @@ export default function timeShift(resp, panel, series) {
     if (/^([+-]?[\d]+)([shmdwMy]|ms)$/.test(series.offset_time)) {
       const matches = series.offset_time.match(/^([+-]?[\d]+)([shmdwMy]|ms)$/);
       if (matches) {
-        const offsetValue = matches[1];
+        const offsetValue = Number(matches[1]);
         const offsetUnit = matches[2];
+
         results.forEach(item => {
           if (_.startsWith(item.id, series.id)) {
-            item.data = item.data.map(row => [moment(row[0]).add(offsetValue, offsetUnit).valueOf(), row[1]]);
+            item.data = item.data.map(row => {
+              const timeWithOffset = row[0] + moment.duration(offsetValue, offsetUnit).valueOf();
```

Comment:
@timroes  I think adding 'offset' looks better than trying to convert `row[0]` to the moment object (without setting the timezone) and only after that setting the offset value.

Reply:
Thanks!